primefaces/src/main/java/org/primefaces/component/media/MediaRenderer.java

```diff
@@ -119,9 +122,31 @@ protected MediaPlayer resolvePlayer(FacesContext context, Media media) {
         if (media.getPlayer() != null) {
             return MediaPlayerFactory.getPlayer(media.getPlayer());
         }
-        else if (media.getValue() instanceof String) {
+
+        Object value = media.getValue();
+
+        // resolve by content-type
+        if (value instanceof StreamedContent) {
+            Map<String, MediaPlayer> players = MediaPlayerFactory.getPlayers();
+            for (MediaPlayer mp : players.values()) {
```

Comment:
I thought we should use index-loops https://github.com/primefaces/primefaces/blob/master/CONTRIBUTING.md#performance-guidelines

Reply:
we mainly use index loops for looping over components (ArrayLists), to void component-list instances in general and to avoid iterator instances

in this case i just copied code and i dont think HashMap#values will give you a simple ArrayList

Reply:
i will adjust the guidelines a bit